fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/RewriteMultiDistinctByCTERule.java

```diff
@@ -147,26 +151,31 @@ public List<OptExpression> transform(OptExpression input, OptimizerContext conte
                 .filter(kv -> kv.getValue().isDistinct()).map(Map.Entry::getKey).collect(Collectors.toList());
         List<ColumnRefOperator> otherAggregate = aggregate.getAggregations().entrySet().stream()
                 .filter(kv -> !kv.getValue().isDistinct()).map(Map.Entry::getKey).collect(Collectors.toList());
+        List<ColumnRefOperator> groupingKeys = aggregate.getGroupingKeys();
+        boolean hasGroupBy = !groupingKeys.isEmpty();
 
         Map<ColumnRefOperator, ScalarOperator> columnRefMap = Maps.newHashMap();
         LinkedList<OptExpression> allCteConsumes = buildDistinctAggCTEConsume(aggregate, distinctAggList, cteProduce,
                 columnRefFactory, columnRefMap);
+        // For inner join(with group by), distinct aggregate cannot have limitation
+        if (hasGroupBy) {
+            allCteConsumes.forEach(opt -> opt.getOp().setLimit(Operator.DEFAULT_LIMIT));
+        }
```

Comment:
check method ensures no limit in agg. can we delete this line?

Reply:
I think better to remain. This can avoid making wrong output if cte reused again later, though this code not works yet.